src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +191,380 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-        # continually remove the known matches from the candidates
-
-        methods = {
-            "same_true": self.get_same_true,
-            "same_diff_own": self.get_matches_at_diff_ownership,
-            "same_quals": self.get_matches_same_qualifiers_ids,
-            "one_ferc1_opt": self.get_only_ferc1_matches,
-        }
-        # compile the connected dfs in a dictionary
-        matches_dfs = {}
-        candidate_matches = deepcopy(candidate_matches_all)
-        for method in methods:
-            connects_method_df = (
-                methods[method](candidate_matches)
-                .assign(match_method=method)
-            )
-            logger.info(
-                f"Matches for {method}:   {len(connects_method_df)}")
-            candidate_matches = self.remove_matches_from_candidates(
-                candidate_matches, connects_method_df)
-            matches_dfs[method] = connects_method_df
-        # squish all of the known matches together
-        matches_df = pd.concat(matches_dfs.values())
-
-        ###########################################
-        # everything below here is just for logging
-        ###########################################
-        ids_to_match = (candidate_matches_all[
-            candidate_matches_all.record_id_eia_ferc1.notnull()]
-            .record_id_eia_deprish.unique())
-        ids_connected = matches_df.record_id_eia_deprish.unique()
-        ids_no_match = (candidate_matches[
-            candidate_matches.record_id_eia_ferc1.notnull()]
-            .plant_id_pudl.unique())
-        logger.info("Portion of unique depreciation records:")
-        logger.info(
-            f"    Matched:   {len(ids_connected)/len(ids_to_match):.02%}")
-        logger.info(
-            f"    No link:   {len(ids_no_match)/len(ids_to_match):.02%}")
-        logger.info(f"""Connected:
-{matches_df.match_method.value_counts(dropna=False)}""")
-        logger.info(f"""Connection Levels:
-{matches_df.level_deprish.value_counts(dropna=False)}""")
-        logger.debug(f"""Only one ferc1 match levels:
-{matches_dfs['one_ferc1_opt'].level_deprish.value_counts(dropna=False)}""")
-
-        # for debugging return all outputs.. remove when this all feels stable
-        # return candidate_matches_all, candidate_matches, matches_df
-        self.matches_df = matches_df
-        return self.matches_df
-
-
-class Scaler(object):
-    """Scales FERC1 data to matching depreciation records."""
-
-    def __init__(self, match_maker):
+    def allocate(
+            self,
+            to_allocate: pd.DataFrame,
+            ppl: pd.DataFrame
+    ) -> pd.DataFrame:
         """
-        Initialize Scaler.
+        Allocate records w/ larger granularities to sub-component plant-parts.
 
-        TODO: Add explicit arguments for the input matches_df into methods
-        instead of accessing the cached df.
+        Implement both steps 2 and 3 as described in
+        :meth:``PlantPartScaler.execute``.
```

Comment:
Double backtics to make it preformated text. But for "roles" like `meth:` you just want single backticks.

Reply:
ty these rules apparently keep sticking in my head incorrectly